firetweet/src/main/java/org/getlantern/firetweet/activity/support/UpdaterActivity.java

```diff
@@ -0,0 +1,175 @@
+package org.getlantern.firetweet.activity.support;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Color;
+import android.graphics.Typeface;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Button;
+import android.widget.ProgressBar;
+import android.view.Window;
+import android.view.View.OnClickListener;
+import android.view.View;
+
+
+import org.getlantern.firetweet.Constants;
+import org.getlantern.firetweet.R;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.net.URLConnection;
+
+/**
+ * Created by todd on 6/15/15.
+ */
+
+
+public class UpdaterActivity extends Activity implements Constants {
+
+    private UpdaterTask mUpdaterTask;
+    private TextView updateAvailableText;
+    private ProgressBar spinner;
+    private ImageView updateIcon;
+    private static final String LOG_TAG = "UpdaterActivity";
+    private static final String APK_URL = "https://raw.githubusercontent.com/firetweet/downloads/master/firetweet.apk";
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_updater);
+
+        spinner = (ProgressBar)findViewById(R.id.pbHeaderProgress);
```

Comment:
The progress bar is indeed a spinner and gives no progress indication. User will not know the current percentage the new version has been downloaded.